src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void handle(Storage storage, FruitTransaction transaction) {
+        storage.reduceFruitQuantity(transaction.getFruit(), transaction.getQuantity());
```

Comment:
You should check the result balance before saving it in the Storage - it should be positive. Throw a RuntimeException in case the balance is negative. This is one of the requirements from the task checklist.

Reply:
It's done in method reduceFruitQuantity